packages/cashier/src/pages/payment-agent/payment-agent-signup/country-selector/country-selector.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { ResidenceList } from '@deriv/api-types';
+import { Formik, Field, FieldProps } from 'formik';
+import { Autocomplete, DesktopWrapper, MobileWrapper, SelectNative, Loading } from '@deriv/components';
+import { localize } from '@deriv/translations';
+import { useWS } from '../../../../hooks';
+import { TReactChangeEvent } from 'Types';
+import { observer } from 'mobx-react';
+
+type TCountrySelectorProps = {
+    onSelect: (country: ResidenceList[number]) => void;
+    className: string;
+};
+
+type TValues = {
+    country_input: string;
+};
+
+const CountrySelector = ({ onSelect, className }: TCountrySelectorProps) => {
+    const [residence_list, setResidenceList] = React.useState<ResidenceList>();
+    const WS = useWS('residence_list');
+
+    const initial_form_values = {
+        country_input: '',
+    };
+
+    const validateFields = (values: TValues) => {
+        const errors: TValues = {
+            country_input: '',
+        };
+        const { country_input } = values;
+
+        if (!country_input) {
+            errors.country_input = localize('Please select the country of document issuance.');
+        } else if (!residence_list?.find(c => c.text === country_input)) {
+            errors.country_input = localize('Please select a valid country of document issuance.');
+        }
+
+        return errors;
+    };
+
+    const submitCountry = (values: TValues) => {
+        const matching_country = residence_list?.find(c => c.text === values.country_input);
+        if (matching_country) {
+            onSelect(matching_country);
+        }
+    };
+
+    React.useEffect(() => {
+        if (!residence_list) {
+            WS.send();
+            setResidenceList(WS.data);
+        }
+    }, [WS, residence_list]);
+
+    return (
```

Comment:
```suggestion
    if (ws.is_loading) return <Loading />

    if (ws.error) return <EmptyState />
    
    if (ws.data) return <Content />
```
@hamid-deriv Let’s do error handling as well 🙏🏻

Reply:
There is no error handling in the design for this section, so It's out of this PR. I will add it when design is provided.

Reply:
@hamid-deriv Unfortunately our designers never consider the error state in the design :(
But I think as a developer it's our responsibility to still handle all the possible states and ask for the design from the designers.

Let's move forward as it’s out of this PR scope, Maybe later we can have a call with design team and discuss this with them 🙇🏻